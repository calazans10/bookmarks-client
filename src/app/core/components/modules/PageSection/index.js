import React from 'react';
import PropTypes from 'prop-types';
import './index.scss';

const PageSection = ({ title, children }) => (
  <section className="page-section">
    <h1 className="page-section__title">{title}</h1>
    {children}
  </section>
);

PageSection.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default PageSection;
