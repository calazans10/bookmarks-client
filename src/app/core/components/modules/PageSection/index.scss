@import 'styles/utilities';

.page-section {
  text-align: center;

  @media(min-width: 600px) {
    margin-top: rem(150);
    margin-bottom: rem(60);
  }

  @media(min-width: 650px) {
    width: rem(650);
    margin-right: auto;
    margin-left: auto;
  }

  &__title {
    margin-bottom: rem(50);
    font-size: em(34);
    font-weight: 400;
  }

  p {
    line-height: 1.5;
  }
}
