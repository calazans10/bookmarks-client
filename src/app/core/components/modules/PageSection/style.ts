import styled from 'styled-components';
import { em, rem } from 'polished';

export const Container = styled.section`
  text-align: center;

  @media (min-width: 600px) {
    margin-top: ${rem('150px')};
    margin-bottom: ${rem('60px')};
  }

  @media (min-width: 650px) {
    width: ${rem('650px')};
    margin-right: auto;
    margin-left: auto;
  }

  p {
    line-height: 1.5;
  }
`;

export const Title = styled.h1`
  margin-bottom: ${rem('50px')};
  font-size: ${em('34px')};
  font-weight: 400;
`;
