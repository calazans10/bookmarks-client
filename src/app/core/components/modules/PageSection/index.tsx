import React from 'react';
import { Container, Title } from './style';

type PageSectionProps = {
  title: string;
  children: React.ReactNode;
};

const PageSection = ({ title, children }: PageSectionProps) => (
  <Container>
    <Title>{title}</Title>
    {children}
  </Container>
);

export default PageSection;
