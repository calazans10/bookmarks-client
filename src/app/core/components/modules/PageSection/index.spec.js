import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PageSection from './index';

describe('PageSection', () => {
  it('renders without crashing', () => {
    const props = {
      title: 'A title',
      children: <p>Lorem Ipsum</p>,
    };
    const wrapper = shallow(<PageSection {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
