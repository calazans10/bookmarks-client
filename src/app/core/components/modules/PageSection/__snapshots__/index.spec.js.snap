// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection renders without crashing 1`] = `
<section
  className="page-section"
>
  <h1
    className="page-section__title"
  >
    A title
  </h1>
  <p>
    Lorem Ipsum
  </p>
</section>
`;
